// ionic-progress-bar/dist/providers/progress-bar/progress-bar.js
import { Injectable } from '@angular/core';
import { ModalController, Events } from 'ionic-angular';
import { SimpleProgressBarComponent } from '../../components/progress-bar/progress-bar';
import { Observable } from 'rxjs/Rx';
var SimpleProgressBarProvider = (function () {
    function SimpleProgressBarProvider(modalCtrl, eventCtrl) {
        this.modalCtrl = modalCtrl;
        this.eventCtrl = eventCtrl;
    }
    SimpleProgressBarProvider.prototype.present = function (message, milestones) {
        var _this = this;
        return Observable.create(function (observer) {
            _this._profile_modal = _this.modalCtrl.create(SimpleProgressBarComponent, { message: message, total: milestones, actual: 0 });
            _this._profile_modal.present().then(function () {
                observer.next();
                observer.complete();
            }).catch(function (err) { return observer.error(err); });
        });
    };
    SimpleProgressBarProvider.prototype.setMilestone = function (message) {
        this.eventCtrl.publish('milestone:add', { message: message }, Date.now());
    };
    SimpleProgressBarProvider.prototype.dismiss = function () {
        if (this._profile_modal) {
            this._profile_modal.dismiss();
        }
    };
    SimpleProgressBarProvider.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    SimpleProgressBarProvider.ctorParameters = function () { return [
        { type: ModalController, },
        { type: Events, },
    ]; };
    return SimpleProgressBarProvider;
}());
export { SimpleProgressBarProvider };
