// ionic-progress-bar/dist/ionic-progress-bar.module.js
import { NgModule } from '@angular/core';
import { IonicModule } from 'ionic-angular';
import { SimpleProgressBarProvider } from './providers/progress-bar/progress-bar';
import { SimpleProgressBarComponent } from './components/progress-bar/progress-bar';
var IonicSimpleProgressBarModule = (function () {
    function IonicSimpleProgressBarModule() {
    }
    IonicSimpleProgressBarModule.forRoot = function () {
        return {
            ngModule: IonicSimpleProgressBarModule,
            providers: [SimpleProgressBarProvider]
        };
    };
    IonicSimpleProgressBarModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        IonicModule
                    ],
                    declarations: [SimpleProgressBarComponent],
                    exports: [SimpleProgressBarComponent],
                    entryComponents: [
                        SimpleProgressBarComponent
                    ],
                    providers: [
                        SimpleProgressBarProvider
                    ]
                },] },
    ];
    return IonicSimpleProgressBarModule;
}());
export { IonicSimpleProgressBarModule };
